### PR TITLE
Fixes 410: generic smtp mail service added

### LINKED ISSUE #410: extend mail service to smtp2go
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

Now yaydoc support only sendgrid.

### Expected behaviour

extend the yaydoc support to smtp2go

### PR DESCRIPTION
**Pull request by sch00lb0y**
## Description

generic smtp mail service implemented
## Related Issue

close #410

## Screenshots (if appropriate):
http://demo08.herokuapp.com/
## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by sch00lb0y** [comment]
@imujjwal96 @pri22296 please review http://demo08.herokuapp.com/

File: backend/mailer.js [hunk 1]
```diff
@@ -6,17 +6,28 @@ var jade = require('jade');
...
-  service: 'SendGrid',
-  auth: {
-    api_user: process.env.SENDGRID_USERNAME,
-    api_key: process.env.SENDGRID_PASSWORD
-  }
-};
+var client;
+if (process.env.SMTP === "true") {
+  client = nodemailer.createTransport({
+    host: process.env.SMTP_HOST,
```
Comment (imujjwal96) [inline backend/mailer.js]: can we not use sendgrid like this?

Comment (sch00lb0y) [reply backend/mailer.js]: I don't think so, sendgrid need api keys. but then here we sending using smtp username and password.

Comment (imujjwal96) [reply backend/mailer.js]: Think harder. https://sendgrid.com/blog/sending-email-nodemailer-sendgrid/

**Comment by sch00lb0y** [comment]
@imujjwal96 I have updated

**Comment by imujjwal96** [comment]
@sch00lb0y Update app.json and kubernetes/yamls/yaydoc/configmap.yaml

**Comment by sch00lb0y** [comment]
@imujjwal96 I have updated. please check. :smile:

**Comment by imujjwal96** [comment]
Update the email to info@yaydoc.org

**Comment by imujjwal96** [comment]
Update this as well: https://github.com/fossasia/yaydoc/blob/master/docs/installation/heroku.md

**Comment by imujjwal96** [comment]
Remove https://github.com/fossasia/yaydoc/blob/development/package.json#L23

**Comment by imujjwal96** [comment]
ICYNI: List of services with their hosts and ports https://github.com/nodemailer/nodemailer/blob/940f6e9a2178ed4861757b389257a22e72885b1c/lib/well-known/services.json

**Comment by sch00lb0y** [comment]
@imujjwal96 updated.

**Comment by imujjwal96** [comment]
The travis build is failing

**Comment by sch00lb0y** [comment]
@imujjwal96 fixed

**Pull request merged**